### ISSUE TITLE
Refund created for uncompleted payments

### ISSUE THREAD

**Issue by bitberry-dev** [solidusio/solidus#2926]
### Steps to reproduce

1. Create order with check payment method.
2. Create refund in admin -> order -> payments.
![2018-10-26 17 55 34](https://user-images.githubusercontent.com/3278464/47574686-cfae9d00-d948-11e8-8f12-5e3a015424a9.png)

### Expected behavior

I expect error when creating refund from uncompleted payment.
Also do not show refund icon for uncompleted payments and do not run refunds#new action.

### Actual behavior

Refund created for payment in "invalid" state.
![2018-10-26 17 55 58](https://user-images.githubusercontent.com/3278464/47574691-d5a47e00-d948-11e8-8935-02cb0df0750c.png)

### System configuration
**Solidus Version**:

"2.8.0.alpha"
master, latest commit 622aa25e6c351c4de16af87e9cba70de596ad959

Is this the right behavior? If not I can fix it.
Also refund model doesn't validate amount on updates.

**Comment by kennyadsl** [comment]
I don't think it is expected, it does not make sense to refund a payment that never happened. Feel free to submit a patch, thanks!

**Comment by jarednorman** [comment]
This was half-fixed by #6094.
